### ISSUE TITLE
wildcard domain support with `--tls-san`

### ISSUE THREAD

**Issue by huapox** [k3s-io/k3s#7821]
wildcard domain support with `--tls-san`, like `--tls-san=*.k8s.local --tls-san=*.xxx.com`

**Comment by brandond** [comment]
This would need to be handled in https://github.com/rancher/dynamiclistener

**Comment by Eshaan-Lumba** [comment]
Hi @brandond, now that rancher/dynamiclistener#272 is merged, wanted to know which k3s release picking up the dynamiclistener bump could be expected to land in?

Thanks!

**Comment by brandond** [comment]
not currently planned, PR welcome

**Comment by Eshaan-Lumba** [comment]
Thanks @brandond. I'll open the k3s PR to bump the `dynamiclistener` dependency on `master`.

One thing to confirm: #272 (wildcard SAN) is currently only on `main` → `v0.9.0-rc.2`, which moves the dep set to wrangler `v3.7.0-rc.1` / k8s `v0.36` / go `1.26`. k3s `master` is on dynamiclistener `v0.7.5` + wrangler `v3.4.0`.

Do you want k3s to move onto the v0.9 line for this (and is a stable `v0.9.0` / wrangler `v3.7.0` planned I can target), or would you prefer #272 backported to a stable line (e.g. a `v0.8.x` on wrangler `v3.6.0`) so k3s avoids an RC dependency?

**Comment by brandond** [comment]
it is probably OK to bump wrangler on main and release-1.36 branches.

If you want to backport this to release-1.35 and older, you'll have to backport the dynamiclistener change to the corresponding Wrangler release branches.

**Comment by Eshaan-Lumba** [comment]
Sounds good, created the PR here for main: https://github.com/k3s-io/k3s/pull/14280, will create a follow up PR for release-1.36 after as well

**Comment by mdrahman-suse** [comment]
Hi @brandond, I am not seeing other backports, is this only for 1.36 and newer?

**Comment by mdrahman-suse** [comment]
# Validated on main branch with commit 0cd872ed

Step: 
Install k3s: `$ curl -sfL https://get.k3s.io | sudo INSTALL_K3S_VERSION=v1.36.2+k3s1 sh -s - --tls-san="*.k8s.local" --write-kubeconfig-mode 0644 --cluster-init --debug`

### Validation
```console
$ k3s -v
k3s version v1.36.2+k3s-0cd872ed (0cd872ed)
go version go1.26.4

$ kubectl get secret k3s-serving -n kube-system -o jsonpath='{.data.tls\.crt}' | base64 -d | openssl x509 -text -noout | grep -A 1 "Subject Alternative Name"
warning: GOCOVERDIR not set, no coverage data emitted
            X509v3 Subject Alternative Name:
                DNS:*.k8s.local, DNS:<redacted>, DNS:kubernetes, DNS:kubernetes.default, DNS:kubernetes.default.svc, DNS:kubernetes.default.svc.cluster.local, DNS:localhost, IP Address:<redacted>, IP Address:<redacted>, IP Address:<redacted>, IP Address:0:0:0:0:0:0:0:1

$ sudo curl --cacert /var/lib/rancher/k3s/server/tls/server-ca.crt   --resolve alpha.k8s.local:6443:127.0.0.1   https://alpha.k8s.local:6443/ping
pong
```

**Comment by brandond** [comment]
@mdrahman-suse yes, 1.36 and newer only; the enhancement to dynamiclistener has not been backported to older minors and there are no plans to do so at this time.
